### PR TITLE
TIG-1328 Migrate all Actors from previous metrics types to metrics::Operation

### PR DESCRIPTION
**Pull request by vrachev**
Please check that I correctly deprecated the previous metrics types. 

Some existing genny classes are still using metrics::registry and metrics::guage (such as orchestrator.hpp). Not sure if we'll want to do that at some point or not. But all of the actors in cast_core are now using metrics::operation.

### CODE REVIEW

File: src/gennylib/include/gennylib/Orchestrator.hpp [hunk 1]
```diff
@@ -107,7 +107,7 @@ class Orchestrator {
     PhaseNumber _current = 0;
 
     // _phaseNumberGauge should keep up to date with _current.
-    metrics::Gauge _phaseNumberGauge;
+    metrics::v1::Gauge _phaseNumberGauge;
```
Comment (rtimmons) [inline src/gennylib/include/gennylib/Orchestrator.hpp]: Let's just kill `_phaseNumberGauge`. We'll probably want something similar in the future but I think it really wants to be a custom gauge within metrics::Operation (requires TIG-1420).

File: src/metrics/include/metrics/metrics.hpp [hunk 2]
```diff
@@ -555,14 +554,14 @@ class Registry {
...
     }
-    Timer timer(const std::string& name) {
-        return Timer{this->_timers[name]};
+    v1::Timer timer(const std::string& name) {
+        return v1::Timer{this->_timers[name]};
     }
-    Gauge gauge(const std::string& name) {
-        return Gauge{this->_gauges[name]};
+    v1::Gauge gauge(const std::string& name) {
+        return v1::Gauge{this->_gauges[name]};
```
Comment (rtimmons) [inline src/metrics/include/metrics/metrics.hpp]: Let's mark these and the associated methods in `ActorContext` as deprecated - see [this SO answer](https://stackoverflow.com/questions/295120/c-mark-as-deprecated).

File: src/cast_core/src/MultiCollectionUpdate.cpp [hunk 3]
```diff
@@ -82,8 +83,7 @@ void MultiCollectionUpdate::run() {
 MultiCollectionUpdate::MultiCollectionUpdate(genny::ActorContext& context)
     : Actor(context),
       _rng{context.workload().createRNG()},
-      _updateTimer{context.timer("updateTime", MultiCollectionUpdate::id())},
-      _updateCount{context.counter("updatedDocuments", MultiCollectionUpdate::id())},
+      _updateOp{context.operation("updateOperation", MultiCollectionUpdate::id())},
```
Comment (rtimmons) [inline src/cast_core/src/MultiCollectionUpdate.cpp]: no need for the "operation" suffix and let's use PascalCase. So just call it `context.operation("Update")`

File: src/gennylib/include/gennylib/Orchestrator.hpp (hunk 1 above)
Comment (rtimmons) [reply src/gennylib/include/gennylib/Orchestrator.hpp]: Actually `timer()`, `gauge()` and `counter()` methods on `ActorContext` can probably just be removed now, ya?

File: src/cast_core/src/MultiCollectionUpdate.cpp (hunk 3 above)
Comment (vrachev) [reply src/cast_core/src/MultiCollectionUpdate.cpp]: Currently all of the actors use camelCase in these situations. Do we prefer if they were all PascalCase?

Comment (rtimmons) [reply src/cast_core/src/MultiCollectionUpdate.cpp]: > Do we prefer if they were all PascalCase?

Yes.

Comment (vrachev) [reply src/cast_core/src/MultiCollectionUpdate.cpp]: 👍

File: src/metrics/include/metrics/metrics.hpp (hunk 2 above)
Comment (vrachev) [reply src/metrics/include/metrics/metrics.hpp]: After talking about this on slack with Ryan we decided not to officially deprecate them, as they are still widely used internally and will spam the compile logs.

**Comment by vrachev** [comment]
Ready for another look.

File: src/metrics/include/metrics/metrics.hpp [hunk 4]
```diff
@@ -86,8 +86,9 @@ using gauged_at_time = std::pair<time_point, gauged_type>;
 
 class Reporter;
 // The v1 namespace is here for two reasons:
-// 1) it's a step towards an ABI. These classes are basically the pimpls of the outer classes
-// 2) it prevents auto-completion of metrics::{X}Impl when you really want metrics::{X}
+// 1) it's a step towards an ABI. These classes are basically the pimpls of the outer classes.
+// 2) it prevents auto-completion of metrics::{X}Impl when you really want metrics::{X}.
+// 3) Counter, Timer and Guage have been deprecated. Use Operation instead when writing actors.
```
Comment (rtimmons) [inline src/metrics/include/metrics/metrics.hpp]: nit - this isn't one of the reasons why the v1 namespace is here :) Just put a blank line here and nix the number

```c++
// 2) ...
//
// Note: Counter, Timer, and Gauge...
```

**Pull request merged**